src/test/shared/utilities/timeoutUtils.test.ts

```diff
@@ -54,11 +54,23 @@ describe('timeoutUtils', async () => {
         const checkTimerMs = 50
         it(`correctly reports an elapsed time with a ${marginOfError}ms margin of error`, async () => {
             const longTimer = new timeoutUtils.Timeout(300)
```

Comment:
If we're going to be asserting against time, then it might make sense to inject an (optional) clock to the `Timeout` clock

Reply:
All of the tests in this file could benefit from a mock clock. I could dig into something like [sinon](https://sinonjs.org/) for [fake timers](https://sinonjs.org/releases/v7.4.2/fake-timers/) or [lolex](https://github.com/sinonjs/lolex) to improve these tests. Whether mocked or injected, I'd prefer to do that as a separate PR.

Reply:
I added a mock clock to the tests in this file. The tests run instantaneously now instead of all of the delays. Win. This PR will get the change.

Reply:
Other tests can benefit from this. If this change looks good, I can make similar changes to other tests in separate PRs.